### DISCUSSION TITLE
Add Write-Progress support into Copy-Item, Move-Item, Remove-Item

### DISCUSSION THREAD

**Discussion by goldjg** [PowerShell/PowerShell#16806] [Enhancements]
Could you please consider adding support for progress bars into the `Copy-Item`, `Move-Item` and `Remove-Item` cmdlets?

I find it strange in 2022 that if I want to kick off a copy and get progress in powershell I either need to execute Robocopy or do it myself using `[io.file]::OpenRead()` and   `[io.file]::OpenWrite()` functionality. Or just do the copy in the GUI.

For very large files and/or slow connections this would be really useful both to verify that the process is still executing and to estimate remaining time left.

**Comment by rkeithhill** [comment]
The vast majority of my file operations with the *-Item cmdlets take less than a few seconds, so a progress bar is not warranted.  But I could see adding a parameter `-ShowProgress` when you know the operation is going to take a while.

**Reply by goldjg** [reply]
Agree, and I wouldn’t see it as a difficult change since the code will likely already be elsewhere in the codebase.

**Comment by iSazonov** [comment]
Unfortunately, today's PowerShell implementation does not allow to work effectively with large volumes. Robocopy is great alternative on Windows.

Also PowerShell uses underlying C# or OS API which haven't progress bar support. Emulation of these operations would be decrease performance significantly.
